tools/ncbi_blast_plus/ncbi_makeblastdb.xml

```diff
@@ -60,52 +65,64 @@ $hash_index
 > '$outfile'
     ]]></command>
     <inputs>
-        <param argument="-dbtype" type="select" display="radio" label="Molecule type of input">
-            <option value="prot">protein</option>
-            <option value="nucl">nucleotide</option>
+      <param argument="-dbtype" type="select" display="radio" label="Molecule type of input">
+        <option value="prot">protein</option>
+        <option value="nucl">nucleotide</option>
+      </param>
+      <!-- TODO Allow merging of existing BLAST databases (conditional on the database type)?
+           NOTE Double check the new database would be self contained first
+      -->
+      <!-- Note this is a mandatory parameter - default should be most recent FASTA file -->
+      <conditional name="input_selection">
+        <param name="source" type="select" label="Input is a">
+          <option value="history">Dataset in history</option>
+          <option value="cached">Genome on server</option>
```

Comment:
You need another conditional level here. Selecting a "Genome on server" makes no sense when the user wants to build a protein database.

Reply:
Good point. Maybe expand the existing choice:

```xml
     <param argument="-dbtype" type="select" display="radio" label="Molecule type of input">
        <option value="prot">protein</option>
        <option value="nucl">nucleotide</option>
      </param>
```

to something like this:

```xml
     <param argument="-dbtype" type="select" display="radio" label="Molecule type of input">
        <option value="prot">protein from your history</option>
        <option value="nucl">nucleotide from your history</option>
        <option value="nucl-server">nucleotide from server</option>
      </param>
```

Reply:
True! (:

Reply:
Would be the simplest way, but on second thought has the downside of breaking the 1:1 relationship between `$dbtype` and the command line option. Nested conditionals are slightly more complex, but the cleaner solution.